### PR TITLE
feat: sdk comm update

### PR DESCRIPTION
**Pull request by abretonc7s**
## **Description**

Major PR update that connections related issues with deeplinks.

- `sdk-communication-layer` update to improve connection and speed up key exchange process.
- Optimize SDK init flow in a 3 stage process (1-init, 2-deeplink, 3-resume, 4-reconnect) which allows to prioritize deeplinked connection over reconnection and solve issue on ios mobile (disconnecting socket).
- Fixes Transaction Header with `null` values on batch rpc calls.

## **Related issues**

Fixes: #

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by andreahaku — DISMISSED** [review]
LGTM

**Comment by christopherferreira9** [comment]
**E2E run:** https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/8326d6b1-f79e-4e25-ad61-31dbfd277927

**Testing:** Full regression to all dapps in all platforms. 

**Issues found:**
- There are known issues related to Branch.io universal links and the behaviour of iOS when dealing with deeplinks on iOS 17. These will be fixed outside the Wallet scope.

File: app/core/DeeplinkManager.js [hunk 1]
```diff
@@ -181,6 +182,9 @@ class DeeplinkManager {
         });
       }
     });
+    if (handle && handle.done) {
+      handle.done();
+    }
```
Comment (omridan159) [inline app/core/DeeplinkManager.js:187]:
```suggestion
handle?.done?.();
```

**Review by omridan159 — APPROVED** [review]

**Review by omridan159 — DISMISSED** [review]

**Review by andreahaku — DISMISSED** [review]
LGTM

**Comment by christopherferreira9** [comment]
Testing: Full regression to all dapps in all platforms.

Issues found:

There are known issues related to Branch.io universal links and the behaviour of iOS when dealing with deeplinks on iOS 17. These will be fixed outside the Wallet scope.

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**